File: app/src/main/java/org/fossasia/susi/ai/SnackbarBehavior.java
package org.fossasia.susi.ai;

import android.content.Context;
import android.support.design.widget.CoordinatorLayout;
import android.support.design.widget.Snackbar;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.LinearLayout;

/**
 * Created by rajdeep1008 on 17/10/16.
 */

public class SnackbarBehavior extends CoordinatorLayout.Behavior<ViewGroup>{

    public SnackbarBehavior(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean layoutDependsOn(CoordinatorLayout parent, ViewGroup child, View dependency) {
        return dependency instanceof Snackbar.SnackbarLayout;
    }

    @Override
    public boolean onDependentViewChanged(CoordinatorLayout parent, ViewGroup child, View dependency) {
        CoordinatorLayout.LayoutParams params = (CoordinatorLayout.LayoutParams) child.getLayoutParams();
        params.bottomMargin = parent.getHeight()- (int) dependency.getY();
        child.setLayoutParams(params);
        return true;
    }

}

File: app/src/main/java/org/fossasia/susi/ai/helper/SnackbarBehavior.kt
package org.fossasia.susi.ai.helper

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.view.ViewGroup
import androidx.coordinatorlayout.widget.CoordinatorLayout
import com.google.android.material.snackbar.Snackbar

/**
 * <h1>Helper class for snackbar behaviour.</h1>

 * Created by rajdeep1008 on 17/10/16.
 */
class SnackbarBehavior
/**
 * Instantiates a new Snackbar behavior.

 * @param context the context
 * *
 * @param attrs the attrs
 */
(context: Context, attrs: AttributeSet) : androidx.coordinatorlayout.widget.CoordinatorLayout.Behavior<ViewGroup>(context, attrs) {

    override fun layoutDependsOn(parent: androidx.coordinatorlayout.widget.CoordinatorLayout, child: ViewGroup, dependency: View): Boolean {
        return dependency is Snackbar.SnackbarLayout
    }

    override fun onDependentViewChanged(parent: androidx.coordinatorlayout.widget.CoordinatorLayout, child: ViewGroup, dependency: View): Boolean {
        val params = child.layoutParams as androidx.coordinatorlayout.widget.CoordinatorLayout.LayoutParams
        params.bottomMargin = parent.height - dependency.y.toInt()
        child.layoutParams = params
        return true
    }

    override fun onDependentViewRemoved(parent: androidx.coordinatorlayout.widget.CoordinatorLayout, child: ViewGroup, dependency: View) {
        val params = child.layoutParams
        if (params is androidx.coordinatorlayout.widget.CoordinatorLayout.LayoutParams) params.bottomMargin = 0
        child.layoutParams = params
    }
}
